bin/nimbus-fml.sh

```diff
@@ -0,0 +1,147 @@
+AS_VERSION=v91.0.0
+
+while (( "$#" )); do
+    case "$1" in
+        -o|--output)
+            OUTPUT_DIR=$2
+            shift 2
+            ;;
+        -h|--help)
+            helptext
+            exit 0
+            ;;
+        --) # end argument parsing
+            shift
+            break
+            ;;
+        --*=|-*) # unsupported flags
+            echo "Error: Unsupported flag $1" >&2
+            exit 1
+            ;;
+        *) # preserve positional arguments
+            FML_PATH=$1
+            shift
+            ;;
+    esac
+done
+
+if [ -z $FML_PATH ]; then
+    if [ -z "$SCRIPT_INPUT_FILE_COUNT" ] || [ "$SCRIPT_INPUT_FILE_COUNT" -eq 0 ]; then
+        echo "Error: No input files provided for the Nimbus Feature Manifest."
+        exit 2
+    fi
+    FML_PATH=$SCRIPT_INPUT_FILE_0
+fi
+
+if [ -z "$SOURCE_ROOT" ]; then
+    echo "Error: No \$SOURCE_ROOT defined."
+    echo "Execute this script as a build step in Xcode."
+    exit 2
+fi
+
+if [ -z "$PROJECT" ]; then
+    echo "Error: No \$PROJECT defined."
+    echo "Execute this script as a build step in Xcode."
+    exit 2
+fi
+if [ -z "$MOZ_BUNDLE_DISPLAY_NAME" ]; then
+    echo "Error: No \$MOZ_BUNDLE_DISPLAY_NAME defined."
+    echo "Execute this script as a build step in Xcode."
+    exit 2
+fi
+
+# We create the nimbus-fml directory, which is gitignored, we use -p to make sure this doesn't fail if it already exists
+mkdir -p ${SOURCE_ROOT}/nimbus-fml
+# We now download the nimbus-fml from the github release
+curl -L https://github.com/mozilla/application-services/releases/download/${AS_VERSION}/nimbus-fml.zip --output ${SOURCE_ROOT}/nimbus-fml/nimbus-fml.zip
+# We also download the checksum
+curl -L https://github.com/mozilla/application-services/releases/download/${AS_VERSION}/nimbus-fml.sha256 --output ${SOURCE_ROOT}/nimbus-fml/nimbus-fml.sha256
+pushd ${SOURCE_ROOT}/nimbus-fml
+shasum --check nimbus-fml.sha256
+popd
+
+## Once the FML is downloaded, we need to unzip it, and run the appropriate nimbus-fml that reflects
+## the architecture of the device running this script.
+unzip -o ${SOURCE_ROOT}/nimbus-fml/nimbus-fml.zip -d ${SOURCE_ROOT}/nimbus-fml
+
+## We get the device's architecture
+ARCH=$(uname -m)
+BINARY_PATH=
+if [[ "$ARCH" == 'x86_64' ]]
+then
+    BINARY_PATH=${SOURCE_ROOT}/nimbus-fml/x86_64-apple-darwin/release/nimbus-fml
+elif [[ "$ARCH" == 'arm64' ]]
+then
+    BINARY_PATH=${SOURCE_ROOT}/nimbus-fml/aarch64-apple-darwin/release/nimbus-fml
+else
+    echo "Error: Unsupported architecture. This script can only run on Mac devices running x86_64 or arm64"
+    exit 2
+fi
+
+## The `MOZ_BUNDLE_DISPLAY_NAME` has the name of the scheme
+## we use it as a channel in the Feature Manifest. 
+## We seperate the first word of it, since Fennec builds sometimes have the user name after the scheme
+SCHEME=${MOZ_BUNDLE_DISPLAY_NAME%% *}
+$BINARY_PATH $FML_PATH -o $OUTPUT_DIR/FML.swift ios features --classname NimbusFML --channel $SCHEME
```

Comment:
I think we should standardize the names we use between Android and iOS; this will make it a bit easier to document.

Suggest: could we go with names:

* `FxNimbus` as the class name.
* `FxNimbus.swift` as the filename.
* `nimbus.fml.yaml` for the manifest name.